gcip/core/need.py

```diff
@@ -1,3 +1,39 @@
+"""This module represents the Gitlab CI [needs](https://docs.gitlab.com/ee/ci/yaml/#needs) keyword.
+
+Needs are to create relationships between `gcip.core.job.Job`s and `gcip.core.sequence.Sequence`s, which will
+then be executed as early as all preceding required jobs finished. This relationship ignores the common ordering by stages.
+
+The `Need` class is mostly for internal use, as you can link `gcip.core.job.Job`s as well as `gcip.core.sequence.Sequence`s
```

Comment:
_mostly_ but not _only_. Read the documentation below. This is just a note, that that the user should not struggle around with `Need`s when only connecting jobs and/or sequences together.

Reply:
Yes, I know. But i do not agree with _mostly_ because it is not _mostly_ for internal use. 
It is just to be used either internal or by the user. 
Therefore I would recommend to not write this sentence. 

It is the users responsibility to decide if the user wants to use this feature or not. 
The sentence describes it as: Dear user if you use this class you do not get any help, cause it is _mostly_ for internal use. 

Don't get me wrong. I like the idea of warning the user, but because `Need` is a core function of the GitLab CI we shouldn't say that.

Reply:
I got your point. I'll push a fix soon.

Reply:
Ok, pls check my last commit. The updated documentation should better express when the use of `Need` is required and when optional.